flytekit/image_spec/image_spec.py

```diff
@@ -407,3 +414,12 @@ def _get_builder(cls, builder: str) -> ImageSpecBuilder:
                     f" Please upgrade envd to v0.3.39+."
                 )
         return cls._REGISTRY[builder][0]
+
+
+def validate_container_registry_name(name: str) -> bool:
+    """Validate Docker container registry name."""
+    # Define the regular expression for the registry name
+    registry_pattern = r"^(localhost:\d{1,5}|([a-z\d\._-]+)(:\d{1,5})?)(/[\w\.-]+)*$"
+
+    # Use regex to validate the given name
+    return bool(re.match(registry_pattern, name))
```

Comment:
It's strange that this is not covered according to codecov. (At some point, I want to be able to trust codecov.)